test-integration/docker-compose.yml

```
@@ -0,0 +1,117 @@
+x-common-env-variables: &common-env-variables
+  NETWORK_AUTH_TOKEN: "i_am_very_secret_token"
+  BACALHAU_API_PORT: "1234"
+  MINIO_ROOT_USER: "minioadmin"
+  MINIO_ROOT_PASSWORD: "minioadminpass"
+  AWS_ACCESS_KEY_ID: "minioadmin"
+  AWS_SECRET_ACCESS_KEY: "minioadminpass"
+
+networks:
+  bacalhau-network:
+    driver: bridge
+
+volumes:
+  minio-volume:
+    driver: local
+  registry-volume:
+    driver: local
+
+services:
+  bacalhau-minio-node:
+    image: quay.io/minio/minio
+    container_name: bacalhau-minio-node-container
+    command: server /data --console-address ":9001"
+    volumes:
+      - minio-volume:/data
+    restart: always
+    networks:
+      - bacalhau-network
+    environment: *common-env-variables
+    healthcheck:
+      test: [ "CMD", "curl", "-f", "http://localhost:9000/minio/health/live" ]
+      interval: 1s
+      timeout: 5s
+      retries: 30
+      start_period: 2s
+
+  bacalhau-container-img-registry-node:
+    image: bacalhau-container-img-registry-node-image
+    container_name: bacalhau-container-img-registry-container
+    volumes:
+      - registry-volume:/var/lib/registry
+    restart: always
+    networks:
+      - bacalhau-network
+    environment:
+      REGISTRY_STORAGE_FILESYSTEM_ROOTDIRECTORY: /var/lib/registry
+      REGISTRY_HTTP_ADDR: "0.0.0.0:5000"
+      REGISTRY_HTTP_TLS_CERTIFICATE: "/certs/bacalhau-container-img-registry-node.crt"
+      REGISTRY_HTTP_TLS_KEY: "/certs/bacalhau-container-img-registry-node.key"
+    healthcheck:
+      test: [ "CMD-SHELL", "nc -zv localhost 5000" ]
+      interval: 1s
+      timeout: 5s
+      retries: 30
+      start_period: 2s
+
+  bacalhau-requester-node:
+    image: bacalhau-requester-node-image
+    container_name: bacalhau-requester-node-container
+    networks:
+      - bacalhau-network
+    environment: *common-env-variables
+    depends_on:
+      bacalhau-minio-node:
+        condition: service_healthy
+    privileged: true
+    command:
+      - /bin/bash
+      - -c
+      - |
+        bacalhau config set "node.network.authsecret" "$${NETWORK_AUTH_TOKEN}" && bacalhau serve --port=$${BACALHAU_API_PORT} --node-type requester
```

I think with 1.5 this has now  changed. 

```suggestion
        bacalhau config set "orchestrator.auth.token" "$${NETWORK_AUTH_TOKEN}" && bacalhau serve --orchestrator -c api.port=$${BACALHAU_API_PORT} 
```

---

@udsamani Thanks for pointing this out. 
Btw, is this new change backward compatible ? 
i.e., will the old command still work ? or will it fail with v1.5 ?

---

yes they are backward compatible

---

actually let me take this back. if a config file had the old config keys and format, we migrate them to the new format when you first run a v1.5.0 bacalhau, but if you call `bacalhau config set old_key=value` using v1.5.0 binary, then we fail with:

```
Error: "node.network.authsecret" is not a valid config key
Hint:  Run 'bacalhau config list' for the complete list of valid config keys
```